### ISSUE TITLE
Add color configuration for Web UI to differentiate production and test

### ISSUE THREAD

**Issue by mkleehammer** [rethinkdb/rethinkdb#3803]
I often need to use the admin console for both development and production and it would be great if there was a very visual way of differentiating them so I don't do something catastrophic in production.

For example, perhaps a new color option in the config file that could be used to do something attention getting.  Changing the color of one of the following might be useful:

* Add a bar of color under the sidebar (say a 6px div all the way across)
* Change the background color of the icons on the sidebar.
* Change the color of the server name on the sidebar.

Maybe put the color closer the destructive items like an optional colored bar across the top of the list of tables and the text area in the Data Explorer.

This might go along with the Friendly server names #3029

**Comment by danielmewes** [comment]
Thank you for your suggestion @mkleehammer .

It sounds useful. By "sidebar" you mean the bar at the top of the console right (the one that displays the server currently being connected to, the number of issues, number of servers and the number of tables)?

Probably the easiest way for implementing and using this would be by having an option in the "Web UI Settings". The ones you get by clicking on the small gear between "Logs" and the "Search" field at the top of the UI. Currently those settings get stored in a cookie, which makes the implementation much easier.

The disadvantage would be that if you connect to say a production server from a different computer, through a different URL, or with a different browser, it would again appear in the default colors since the cookie won't be found.

A more complex but probably better solution would be storing the color in the `r.db('rethinkdb').table('cluster_config')` table. This would require an extension to the server code to store and synchronize this setting across nodes in the cluster. Depending on how commonly this feature is going to be used, it has the disadvantage that it adds clutter to the configuration table. It might be worth it though.

@mglukhovsky what is your opinion on this?

**Comment by mglukhovsky** [comment]
If this is a matter of personal preference, a cookie would be sufficient. We would also have to think about how much of the UI you'd be allowed to recolor.

If this is intended to reliably distinguish a production vs development cluster, we would probably have to store it in the system cluster configuration, which would definitely require additional work.

Would it be sufficient if we allowed adding a brightly colored tag to the header of the web UI that allows custom text / color? (e.g. _development_ or _production_)

**Comment by chrisvariety** [comment]
We do this currently, works really well:

![rethinkdb administration console 2015-02-19 13-24-48](https://cloud.githubusercontent.com/assets/44855/6276317/be2df388-b83a-11e4-90d7-85747da0cdad.png)

rethinkdb will automatically use the machine's hostname, so we just set 'db-TI-production' as the hostname and it appears both on the prompt (when sshing in) and in the rethinkdb web UI.

**Comment by mkleehammer** [comment]
I think the cookie would actually be worse since it would sometimes be red (or whatever you choose) but not if you clear your cache or use a different machine.  If the purpose is to identify production, having an indicator that is *sometimes* there would make matters worse.

The host name works ok but when there are multiple machines on AWS or one some other system where hostnames are out of our control and look random it isn't quite as helpful.

**Comment by timmaxw** [comment]
The server's "RethinkDB name" is set to the hostname by default, but you can change it to something else by passing the `--server-name` parameter on the command line or by writing to the `rethinkdb.server_config` system table.

**Comment by chrisvariety** [comment]
Just to be clear, we use AWS and you can set the hostname on AWS no problem... :wink: we also have multiple machines as part of a cluster, but they all get the same hostname currently. rethinkdb automatically appends a random (?) number to the end so our actual name ends up being "db_TI_production_378"

**Comment by danielmewes** [comment]
You could use the cookie to identify development machines (e.g. with a green color). That way you know to be careful when the UI appears in the default color. I agree though that it's not quite ideal.

Like @timmaxw said you can override the RethinkDB server name manually through the system table or even through the instance configuration file in `/etc/rethinkdb/instandces.d/...` by adding a `server-name=PROD_...` line to the file.
... or try what @chrisvariety suggested and setting the host name through AWS.

Would one of those options work for you?

I think the color coding would still be more visible, but is definitely more work to implement.

**Comment by timmaxw** [comment]
Also, we've discussed the possibility of adding a "cluster ID" to make it hard for users to accidentally connect two clusters together. We could also use the cluster ID to solve this problem. If we display the cluster ID at the top of the web UI, then users could set it to "development" or "production" or whatever. Another option would be to choose a color scheme for the web UI by hashing the cluster ID, so different clusters would automatically each have their own color scheme. (But we probably won't implement cluster IDs for a few months at least, so it's not an immediate solution.)

**Comment by mkleehammer** [comment]
I'll try those.  I had tried setting machine-name in the config file but it did not affect the host name.  (I started with server-name like the docs but the 1.15 server reported it did not know this setting.)

Since I assume these techniques work should this be closed?  I think a color would be useful long-term addition - it is much more obvious than text, particularly if you reserve bright colors only for production.  For example, one app I am using has a red navbar if node_env-production and the login is marked as a developer.

**Comment by timmaxw** [comment]
In 1.15 and earlier the parameter was called `machine-name`. In 1.16 it is called `server-name`.

**Comment by danielmewes** [comment]
@mkleehammer I agree that your color idea is useful and that we should definitely keep it in mind. Though if one of the other solutions work this wouldn't be the most high-priority feature.

**Comment by mkleehammer** [comment]
I was finally able to upgrade to 1.16 and renamed the server to "Production" which is working for now.  Red would really be nice because you can't *not* notice it, but this will do well for now.  Thanks.  I didn't know if I was supposed to close this or if you wanted to keep it as a reminder.
